nixos/modules/services/networking/ntp/chrony.nix

```diff
@@ -85,6 +88,33 @@ in
         '';
       };
 
+      enableRTCTrimming = mkOption {
+        type = types.bool;
+        default = true;
+        description = lib.mdDoc ''
+          Enable tracking of the RTC offset to the system clock and automatic trimming.
+          See also [](#opt-services.chrony.autotrimThreshold)
+
+          ::: {.note}
+          This is not compatible with the `rtcsync` directive, which naively syncs the RTC time every 11 minutes.
```

Comment:
This should be validated using an `assertions` clause in the module. There's no reason to let Chrony crash; we should just reject the invalid configuration at `nixos-rebuild build` time.

Reply:
That sounds good, but how do you propose we check this?
Check if extraConfig contains the string "rtcsync" seems a bit error-prone.

Reply:
Oh, that's right, we don't have an `rtcsync` setting... Sorry, it's been a while since I touched a lot of this up. Ugh. Frankly, maybe we should add it too, then make it an assertion? We could in theory use `hasInfix` to find it inside the `extraConfig` too, perhaps...

Reply:
I've thought about adding an `rtcsync` option, but there is no benefit compared to `rtcfile` and `rtcautotrim` on Linux. On any other platfom, there is no support for the latter two options, but this doesn't apply to us here.

Testing the `extraConfig` would probably work if we strip comments and then check for "rtcsync", but I'm not sure how to do it decently.
I might need some help with this if you don't mind! :)

Reply:
Added the assertion, it should only trigger when the statement is actually included.